superset-frontend/packages/superset-ui-core/src/query/types/Column.ts

```diff
@@ -27,6 +27,8 @@ export interface AdhocColumn {
   optionName?: string;
   sqlExpression: string;
   expressionType: 'SQL';
+  columnType?: 'AXIS' | 'SERIES';
```

Comment:
To be more explicit, I think `BASE_AXIS` could be more clear, as `AXIS` could refer to the either axis (it could be argued that `SERIES` is on the `VALUE_AXIS` or similar).
```suggestion
  columnType?: 'BASE_AXIS' | 'SERIES';
```